xtask/src/templates/control_id_rv32im.rs.tmpl

```diff
@@ -27,3 +27,30 @@ pub const POSEIDON2_CONTROL_IDS: ControlIds = [{}];
 
 /// Control IDs for each power-of-two of the rv32im circuit using Blake2b.
 pub const BLAKE2B_CONTROL_IDS: ControlIds = [{}];
+
+#[inline]
+pub const fn control_id_sha256(po2: usize) -> Option<Digest> {{
+    if po2 < risc0_zkp::MIN_CYCLES_PO2 || po2 > risc0_zkp::MAX_CYCLES_PO2  {{
+        None
+    }} else {{
+        Some(SHA256_CONTROL_IDS[po2 - risc0_zkp::MIN_CYCLES_PO2])
+    }}
+}}
+
+#[inline]
+pub const fn control_id_poseidon2(po2: usize) -> Option<Digest> {{
+    if po2 < risc0_zkp::MIN_CYCLES_PO2 || po2 > risc0_zkp::MAX_CYCLES_PO2  {{
+        None
+    }} else {{
+        Some(POSEIDON2_CONTROL_IDS[po2 - risc0_zkp::MIN_CYCLES_PO2])
+    }}
+}}
+
+#[inline]
+pub const fn control_id_blake2b(po2: usize) -> Option<Digest> {{
+    if po2 < risc0_zkp::MIN_CYCLES_PO2 || po2 > risc0_zkp::MAX_CYCLES_PO2  {{
+        None
+    }} else {{
+        Some(BLAKE2B_CONTROL_IDS[po2 - risc0_zkp::MIN_CYCLES_PO2])
+    }}
+}}
```

Comment:
I'm not sure these helpers should live in generated code

Reply:
I was having trouble thinking of a better place to put them. Probably next-best in my mind would be `risc0/circuit/rv32im/src/lib.rs`

Reply:
That seems like a good place. And also it feels like we could write a generic function instead of having one for each hash.

Reply:
I'll go ahead and move them, and also revert the rename of this file.